commit e872b0a8e1dc0ecece01144896f44664eb6914e2
Author: root <root@vm1.sorenson.redhat.com>
Date:   Fri Sep 21 14:43:37 2018 -0500

    fat: set the s_time_gran for msdos or vfat mounts
    
    For vfat, ctime granularity is 10ms; set the super_block's
    s_time_gran to 10ms for vfat and 1 second for msdos mounts.
    
    Signed-off-by: Frank Sorenson <sorenson@redhat.com>

diff --git a/fs/fat/inode.c b/fs/fat/inode.c
index d6b81e3..3607186 100644
--- a/fs/fat/inode.c
+++ b/fs/fat/inode.c
@@ -1626,6 +1626,7 @@ int fat_fill_super(struct super_block *sb, void *data, int silent, int isvfat,
 	sb->s_magic = MSDOS_SUPER_MAGIC;
 	sb->s_op = &fat_sops;
 	sb->s_export_op = &fat_export_ops;
+	sb->s_time_gran = isvfat ? 10000000 : 1000000000;
 	mutex_init(&sbi->nfs_build_inode_lock);
 	ratelimit_state_init(&sbi->ratelimit, DEFAULT_RATELIMIT_INTERVAL,
 			     DEFAULT_RATELIMIT_BURST);
